Update links and headers

bamurtaugh opened the pull request:
- Make the titles in spec.md and reference.md into links, for easier sharing and consistency (other pages on the site use this format)
- Update broken links in https://containers.dev/implementors/spec
- Add @edgonmsft's content from https://github.com/devcontainers/spec/pull/39/files
- Add info on contributing schema hints/IntelliSense
     - Once we're happy with this wording, I can open a PR to update https://github.com/devcontainers/spec/blob/main/contributing.md

cc @Chuxel

Commit:
Update headers to links

Commit:
Links and formatting

Commit:
Contributing schema info

Commit:
Updates from latest spec PR

Chuxel commented:
Made some additional comments on https://github.com/devcontainers/spec/pull/39

For the schema - do we want people doing a PR to VS Code or the spec repo once we do https://github.com/devcontainers/spec/issues/14 ? (and should we do that at this point is another related question)

//cc @chrmarti  on the schema

bamurtaugh commented:
Thanks @Chuxel! I'll keep an eye on that PR and add further updates here.

> For the schema - do we want people doing a PR to VS Code or the spec repo once we do https://github.com/devcontainers/spec/issues/14 ? (and should we do that at this point is another related question)

Ah good point. I think a PR to the spec repo would be best as it's tool agnostic. In which case, moving the schema now makes sense to me, if @chrmarti agrees. This could at least account for part of https://github.com/devcontainers/spec/issues/14, if we needed to wait or discuss a bit more the schema store part.

Do we need to advise users to update anything in VS Code to get sufficient IntelliSense? Based on @chrmarti's [comment](https://github.com/devcontainers/spec/issues/14#issuecomment-1079193181): `One disadvantage might be that other editors will not offer IntelliSense for the VS Code and Codespaces parts, but that might be fine`, it sounds like this would be something for other editors to account for, but not something we'd need to provide updates for in VS Code.

Commit:
Latest PR updates

Commit:
Update contributing

bamurtaugh commented:
Added all the latest from @edgonmsft's PR https://github.com/devcontainers/spec/pull/39.

Regarding the contributions.md update to describe updating the schema: I propose we remove that info until we move the schema to the spec repo (part of https://github.com/devcontainers/spec/issues/14), otherwise it feels less VS Code agnostic to direct people to the VS Code repo. I've removed that content now, and I'll proceed with merging this PR.
